mlrun/db/httpdb.py

```diff
@@ -140,6 +144,18 @@ def get_base_api_url(self, path: str, version: str = None) -> str:
         url = f"{self.base_url}/{path_prefix}/{path}"
         return url
 
+    def request_with_retry(self, method, url, **kwargs):
+        retry_count = 0
+        while True:
+            try:
+                response = self.session.request(method, url, **kwargs)
+                return response
+            except HTTP_RETRY_EXCEPTIONS as exc:
+                retry_count += 1
+                if retry_count >= HTTP_RETRY_AMOUNT:
+                    raise exc
+                time.sleep(HTTP_RETRY_BACKOFF)
```

Comment:
IIRC when the adapter catches error from the `status_forcelist`, it gathers the errors and then when passing the retry amount it attaches the log from the other retries or how many retries it had.

Reply:
So I think we should add a debug / warning log when we decide to raise the exception